sklearn/linear_model/base.py

```diff
@@ -492,7 +501,16 @@ def fit(self, X, y, sample_weight=None):
             # Sample weight can be implemented via a simple rescaling.
             X, y = _rescale_data(X, y, sample_weight)
 
-        if sp.issparse(X):
+        if self.positive:
+            if y.ndim < 2:
+                self.coef_, self._residues = optimize.nnls(X, y)
```

Comment:
either you make _residues a proper attribute or don't attach it to the object. You don't use it anyway later or in the tests.